pkg/kubectl/resource/builder.go

```diff
@@ -622,11 +622,17 @@ func (b *Builder) resourceMappings() ([]*meta.RESTMapping, error) {
 		return nil, fmt.Errorf("you may only specify a single resource type")
 	}
 	mappings := []*meta.RESTMapping{}
+	set := sets.NewString()
 	for _, r := range b.resources {
 		mapping, err := b.mappingFor(r)
 		if err != nil {
 			return nil, err
 		}
+		// This ensures the mappings for resources(shortcuts, plural) unique
+		if set.Has(mapping.Resource) {
```

Comment:
is Resource fully qualified? should we work off of mapping.GroupVersionKind instead?

Reply:
I checked the mappings, resource works but seems GroupVersionKind is more qualified, will update to use that instead, thanks!